addons/utm/models/utm_campaign.py

```diff
@@ -25,6 +27,36 @@ class UtmCampaign(models.Model):
     is_auto_campaign = fields.Boolean(default=False, string="Automatically Generated Campaign", help="Allows us to filter relevant Campaigns")
     color = fields.Integer(string='Color Index')
 
+    _sql_constraints = [
+        ('unique_name', 'UNIQUE(name)', 'The name must be unique'),
+    ]
+
+    @api.depends('title')
+    def _compute_name(self):
+        for campaign in self:
+            if not campaign.id and campaign.title:
+                # Compute the name only if the campaign is not yet saved
+                campaign.name = self.env['utm.mixin']._set_name_unique(self._name, campaign.title)
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        created_names = []
+        for vals in vals_list:
+            if not vals.get('title'):
+                vals['title'] = vals.get('name')
+
+            force_uuid = vals.get('name') in created_names
+            vals['name'] = self.env['utm.mixin']._set_name_unique(self._name, vals.get('name'), force_uuid=force_uuid)
+            created_names.append(vals['name'])
+
+        return super(UtmCampaign, self).create(vals_list)
+
+    def copy(self, default=None):
```

Comment:
Note that if we set copy=False, it should go through the compute method that already does the same thing... isn't it ?